src/parsers/power_models_data.jl

```diff
@@ -273,52 +287,55 @@ function read_loads!(sys::System, data, bus_number_to_bus::Dict{Int, ACBus}; kwa
     end
 end
 
-function make_loadzone(name, active_power, reactive_power; kwargs...)
-    _get_name = get(kwargs, :loadzone_name_formatter, _get_pm_dict_name)
+function make_loadzone(
+    name::String,
+    active_power::Float64,
+    reactive_power::Float64;
+    kwargs...,
+)
     return LoadZone(;
         name = name,
-        peak_active_power = sum(active_power),
-        peak_reactive_power = sum(reactive_power),
+        peak_active_power = active_power,
+        peak_reactive_power = reactive_power,
     )
 end
 
 function read_loadzones!(sys::System, data, bus_number_to_bus::Dict{Int, ACBus}; kwargs...)
     @info "Reading LoadZones data in PowerModels dict to populate System ..."
-
+    _get_name = get(kwargs, :loadzone_name_formatter, _get_pm_dict_name)
     zones = Set{Int}()
-    for (i, bus) in data["bus"]
+    zone_bus_map = Dict{Int, Vector}()
+    for (_, bus) in data["bus"]
         push!(zones, bus["zone"])
+        push!(get!(zone_bus_map, bus["zone"], Vector()), bus)
     end
 
+    load_zone_map =
+        Dict{Int, Dict{String, Float64}}(i => Dict("pd" => 0.0, "qd" => 0.0) for i in zones)
+    for (key, load) in data["load"]
+        zone = data["bus"][load["load_bus"]]["zone"]
```

Comment:
You could remove quite a few dictionary lookups.
```suggestion
        zone = load_zone_map[data["bus"][load["load_bus"]]["zone"]]
```
And then perform the sums together.

Reply:
Why not take this suggestion?